src/main/java/org/terasology/anatomy/ui/AnatomyScreenWindow.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.anatomy.ui;
+
+import org.terasology.anatomy.event.AnatomyStatusGatheringEvent;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.characters.CharacterComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.rendering.nui.BaseInteractionScreen;
+import org.terasology.rendering.nui.databinding.ReadOnlyBinding;
+import org.terasology.rendering.nui.skin.UISkin;
+import org.terasology.rendering.nui.widgets.UILabel;
+import org.terasology.utilities.Assets;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public class AnatomyScreenWindow extends BaseInteractionScreen {
+    private static final String ANATOMY_PART_PREFIX = "Anatomy:";
+    private EntityRef player = EntityRef.NULL;
+
+    private final UISkin greenTextSkin = Assets.getSkin("Anatomy:greenText").get();
+    private final UISkin redTextSkin = Assets.getSkin("Anatomy:redText").get();
+
+    @Override
+    public void initialise() {
+    }
+
+    @Override
+    protected void initializeWithInteractionTarget(EntityRef interactionTarget) {
+    }
+
+    private void reInit() {
+        player = CoreRegistry.get(LocalPlayer.class).getCharacterEntity();
+
+        // In case the player has been created yet, exit out early to prevent an error.
+        if (player == EntityRef.NULL) {
+            return;
+        }
+    }
+
+    @Override
+    public void onOpened() {
+        EntityRef characterEntity = CoreRegistry.get(LocalPlayer.class).getCharacterEntity();
+        CharacterComponent characterComponent = characterEntity.getComponent(CharacterComponent.class);
+
+        // In case the player has been created yet, exit out early to prevent an error.
+        if (characterComponent == null) {
+            return;
+        }
+
+        // If the reference to the player entity hasn't been set yet, or it refers to a NULL entity, call the reInit()
+        // method to set it. The getId() check is necessary for certain network entities whose ID is 0, but are
+        // erroneously marked as existent.
+        if (!player.exists() || (player.exists() && (player == EntityRef.NULL || player.getId() == 0 || player == null))) {
```

Comment:
That was taken from my work in [`CharacterScreenWindow`](https://github.com/Terasology/Equipment/blob/develop/src/main/java/org/terasology/equipment/ui/CharacterScreenWindow.java) in the `Equipment` module. It was needed to prevent an odd scenario where the player could be marked as existent, when actually it should have been destroyed.